docs/proposals/docker-event-collection.md

```diff
@@ -0,0 +1,90 @@
+# Docker Event Collection
+## Overview
+Docker events are not collected or stored by the Openshift cluster.  Certain customers are [required](https://trello.com/c/qxoFsKz9/) to audit these events (e.g. security).  This document is the architecture and design proposal for gathering Docker events for persistence as log entries interleaved with other operational logs.
+
+## Requirements
+### Security 
+Docker event log entries shall have the same visibility as other operational log entries.
+### Storage and Retention
+Docker events shall be stored for the same retention period as other log event messages.  The default curation period is 31 days.  The default indices are the operations indices
+
+# Proposal
+Openshift is deployed with [ProjectAtomic Docker](https://github.com/projectatomic/docker/blob/docker-1.12.6/api/server/middleware/audit_linux.go) that enhances audit logging of container events.  These events are logged to the audit file on the node.  The log collector for the logging stack already has access to the file system and will collect docker events from this file.  
+
+**NOTE:** Only Docker Events will be parsed from the audit log; all others will be dropped.
+
+An example event messages written to `/var/log/audit/messages`:
+
+```
+type=VIRT_CONTROL msg=audit(1504273220.029:520188): pid=14500 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:container_runtime_t:s0 msg='vm=docker.io/openshift/origin-logging-kibana@sha256:a9d309679
+cd8c1b098c696b8acef9fba401c09ed7568ad266eb32208c2ce4da9 user=? auid=4294967295 reason=api op=exec vm-pid=17235 exe=container-entrypoint hostname=logging-kibana-1-xxmr3  exe="/usr/bin/dockerd-current" hostname=? 
+addr=? terminal=? res=success'
+
+```
+
+The additional fields:
+
+|Field|Type|Definition| Example |
+|----|-----|-|------|
+|op| string|The operation or action being performed|exec
+|reason | string|The constant value 'api'|api
```

Comment:
given the `reason` is constant, would we maybe want this to be implicit? Is it possible that this will carry other values in future or is there another reason to have this explicitly stated?

Reply:
I don't actually know if this value will ever change.  I presume you could have some event generated that is not related to an API call and maybe this is the source of this field.  Do you think maybe it would be better to simply not identify this field in the proposal but to persist it to storage?  Or maybe for the sake of disk savings we should exclude it.  I don't know what value it really provides.

Reply:
> I presume you could have some event generated that is not related to an API call and maybe this is the source of this field

Then I think you were correct to include it here if it could potentially change in future and provide some unique information.